modules/flowable-ui-modeler/flowable-ui-modeler-logic/src/main/java/org/flowable/app/service/editor/FlowableModelQueryService.java

```diff
@@ -150,121 +158,240 @@ public ResultListDataRepresentation getModelsToIncludeInAppDefinition() {
+    String description = process.getDocumentation();
+
+    ModelRepresentation model = new ModelRepresentation();
+    model.setKey(process.getId());
+    model.setName(name);
+    model.setDescription(description);
+    model.setModelType(AbstractModel.MODEL_TYPE_BPMN);
+
+    List<Model> existingProcessModels = modelRepository
+        .findByKeyAndType(process.getId(), AbstractModel.MODEL_TYPE_BPMN);
+
+    Model modelToReturn;
+    if (existingProcessModels.size() > 0)
+    {
+      modelToReturn = existingProcessModels.get(0);
+      modelService.saveModel(modelToReturn, modelNode.toString(), null, true, "imported",
+          SecurityUtils.getCurrentUserObject());
+    }
+    else
+    {
+      modelToReturn = modelService
+          .createModel(model, modelNode.toString(), SecurityUtils.getCurrentUserObject());
+    }
+    return new ModelRepresentation(modelToReturn);
+  }
+
+  private static class BpmnFileException extends Exception
+  {
+    public BpmnFileException(String message)
+    {
+      super(message);
+    }
 
-            } catch (Exception e) {
-                logger.error("Import failed for {}", fileName, e);
-                throw new BadRequestException("Import failed for " + fileName + ", error message " + e.getMessage());
-            }
-        } else {
-            throw new BadRequestException("Invalid file name, only .bpmn and .bpmn20.xml files are supported not " + fileName);
-        }
+    BpmnFileException(String message, Throwable cause)
+    {
+      super(message, cause);
+    }
+  }
+
+  private static class BpmnFileValidationException extends Exception
+  {
+    private Map<String, Object> validationErrors;
+
+    BpmnFileValidationException(Map<String, Object> validationErrors)
+    {
+      super("There are validation errors!");
+      this.validationErrors = validationErrors;
+    }
+  }
+
+  private void assertFileType(String fileName) throws BpmnFileValidationException
+  {
+    //file "endings" because they aren't really an extension
+    List<String> validFileEndings = Arrays.asList(BpmnFile.BPMN20_XML_ENDING, BpmnFile.BPMN_EXT);
+
+    if (StringUtils.isBlank(fileName) || (StringUtils.isNotBlank(fileName) && validFileEndings.stream()
+        .noneMatch(fileName::endsWith)))
+    {
+      throw new BpmnFileValidationException(Collections.singletonMap("message",
+          "Invalid file name [" + fileName + "], only the following file types are supported (" + StringUtils
+              .join(validFileEndings, ", ") + ")."));
     }
+  }
 
-    protected ModelRepresentation createModelRepresentation(AbstractModel model) {
-        ModelRepresentation representation = null;
-        if (model.getModelType() != null && model.getModelType() == 3) {
+  protected ModelRepresentation createModelRepresentation(AbstractModel model)
+  {
+    ModelRepresentation representation;
+    if (model.getModelType() != null && model.getModelType() == AbstractModel.MODEL_TYPE_APP)
```

Comment:
Remove magic number, just want to check if this is the correct constant...

Reply:
looks good to me